### PR TITLE
All native_dropout interface to handle inference.

### PR DESCRIPTION
**Pull request by kevinstephano**
Fixes #{issue number}

### CODE REVIEW

File: aten/src/ATen/native/cuda/Dropout.cu [hunk 1]
```diff
@@ -326,7 +326,9 @@ inline void launcher(
 
 std::tuple<Tensor,Tensor>
 native_dropout_cuda(const Tensor& self, double p, double scale, bool train){
-  TORCH_CHECK(train, "Train parameter is incorrectly set!");
+  if (!train) {
+    return std::tuple<Tensor, Tensor>(self, at::empty(0, self.options().dtype(kBool)));
```
Comment (jjsjann123) [inline aten/src/ATen/native/cuda/Dropout.cu:330]: For the record, this doesn't look right with returning an empty mask. Backward doesn't take training flag, so we would want mask to be properly set.

But let's not worry much about it here. If it works in your example, it's good enough to be pushed to our branch.
I'm fixing similar issues on upstream dropout PR.

**Review by jjsjann123 — APPROVED** [review]
Good enough. :stuck_out_tongue_closed_eyes:

**Comment by csarofeen** [comment]
@kevinstephano  @jjsjann123 should the conflicts be fixed and should we merge this?

**Comment by jjsjann123** [comment]
> @kevinstephano @jjsjann123 should the conflicts be fixed and should we merge this?

I don't think we need this any more. upstream dropout PR should have fixed it. already

**Pull request closed without merging**